src/bindings/c/include/openvino/c/ov_infer_request.h

```diff
@@ -70,37 +71,149 @@ OPENVINO_C_API(ov_status_e)
 
 /**
- * @brief Gets an output tensor to infer on.
+ * @brief Gets an input/output tensor by const port.
  * @ingroup infer_request
  * @param infer_request A pointer to the ov_infer_request_t.
- * @param idx Index of the tensor to get.
+ * @param port Port of the tensor to get. @p port is not found, an exception is thrown.
  * @param tensor Reference to the tensor.
  * @return Status code of the operation: OK(0) for success.
  */
 OPENVINO_C_API(ov_status_e)
-ov_infer_request_get_output_tensor(const ov_infer_request_t* infer_request, size_t idx, ov_tensor_t** tensor);
+ov_infer_request_get_tensor_by_const_port(const ov_infer_request_t* infer_request, const ov_output_const_node_t* port, ov_tensor_t** tensor);
+
+/**
+ * @brief Gets an input/output tensor by port.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param port Port of the tensor to get. @p port is not found, an exception is thrown.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_get_tensor_by_port(const ov_infer_request_t* infer_request, const ov_output_node_t* port, ov_tensor_t** tensor);
+
+/**
+ * @brief Gets an input tensor by the index of input tensor.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param idx Index of the tensor to get. @p idx. If the tensor with the specified @p idx is not found, an exception
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_get_input_tensor_by_index(const ov_infer_request_t* infer_request, const size_t idx, ov_tensor_t** tensor);
+
+/**
+ * @brief Gets an input tensor from the model with only one input tensor.
+ * @ingroup infer_request
+ * @note If model has several inputs, an exception is thrown.
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_get_input_tensor(const ov_infer_request_t* infer_request, ov_tensor_t** tensor);
+
+/**
+ * @brief Gets an output tensor by the index of output tensor.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param idx Index of the tensor to get. @p idx. If the tensor with the specified @p idx is not found, an exception
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_get_output_tensor_by_index(const ov_infer_request_t* infer_request, const size_t idx, ov_tensor_t** tensor);
+
+/**
+ * @brief Gets an output tensor from the model with only one output tensor.
+ * @note f model has several outputs, an exception is thrown.
```

Comment:
```suggestion
 * @note if model has several outputs, an exception is thrown.
```
I think we don't throw exceptions in C API

Reply:
ok, will update.

Reply:
yes, this exception will be caught by C API and will not expose to C user.

Reply:
done